crates/ruff_python_parser/tests/snapshots/invalid_syntax@expressions__dict__double_star.py.snap

```diff
@@ -0,0 +1,641 @@
+5 | {a: 1, **x if True else y}
+6 | {**lambda x: x, b: 2}
+  |
+
+
+  |
+2 | # the ones which are higher than that.
+3 | 
+4 | {**x := 1}
+  |          ^ Syntax Error: expected Colon, found Rbrace
+5 | {a: 1, **x if True else y}
+6 | {**lambda x: x, b: 2}
+  |
+
+
+  |
+4 | {**x := 1}
+5 | {a: 1, **x if True else y}
+  |            ^^ Syntax Error: expected Comma, found If
+6 | {**lambda x: x, b: 2}
+7 | {a: 1, **x or y}
+  |
+
+
+  |
+4 | {**x := 1}
+5 | {a: 1, **x if True else y}
+  |                          ^ Syntax Error: Expected a statement
+6 | {**lambda x: x, b: 2}
+7 | {a: 1, **x or y}
+  |
+
+
+  |
+4 | {**x := 1}
+5 | {a: 1, **x if True else y}
+  |                           ^ Syntax Error: Expected a statement
+6 | {**lambda x: x, b: 2}
+7 | {a: 1, **x or y}
+8 | {**x and y, b: 2}
+  |
+
+
+  |
+4 | {**x := 1}
+5 | {a: 1, **x if True else y}
+6 | {**lambda x: x, b: 2}
+  |    ^^^^^^^^^^^ Syntax Error: lambda expression cannot be used here
+7 | {a: 1, **x or y}
+8 | {**x and y, b: 2}
+  |
+
+
+  |
+5 | {a: 1, **x if True else y}
+6 | {**lambda x: x, b: 2}
+7 | {a: 1, **x or y}
+  |            ^^ Syntax Error: expected Comma, found Or
+8 | {**x and y, b: 2}
+9 | {a: 1, **not x, b: 2}
+  |
+
+
+  |
+5 | {a: 1, **x if True else y}
+6 | {**lambda x: x, b: 2}
+7 | {a: 1, **x or y}
+  |                ^ Syntax Error: expected Colon, found Rbrace
+8 | {**x and y, b: 2}
+9 | {a: 1, **not x, b: 2}
+  |
+
+
+   |
+ 6 | {**lambda x: x, b: 2}
+ 7 | {a: 1, **x or y}
+ 8 | {**x and y, b: 2}
+   |      ^^^ Syntax Error: expected Comma, found And
+ 9 | {a: 1, **not x, b: 2}
+10 | {**x in y}
+   |
+
+
+   |
+ 6 | {**lambda x: x, b: 2}
+ 7 | {a: 1, **x or y}
+ 8 | {**x and y, b: 2}
+   |           ^ Syntax Error: expected Colon, found Comma
+ 9 | {a: 1, **not x, b: 2}
+10 | {**x in y}
+   |
+
+
+   |
+ 7 | {a: 1, **x or y}
+ 8 | {**x and y, b: 2}
+ 9 | {a: 1, **not x, b: 2}
+   |          ^^^^^ Syntax Error: unary `not` expression cannot be used here
+10 | {**x in y}
+11 | {**x not in y}
+   |
+
+
+   |
+ 8 | {**x and y, b: 2}
+ 9 | {a: 1, **not x, b: 2}
+10 | {**x in y}
+   |      ^^ Syntax Error: Expected an expression or a '}'
+11 | {**x not in y}
+12 | {**x < y}
+   |
+
+
+   |
+ 8 | {**x and y, b: 2}
+ 9 | {a: 1, **not x, b: 2}
+10 | {**x in y}
+   |          ^ Syntax Error: expected Colon, found Rbrace
+11 | {**x not in y}
+12 | {**x < y}
+   |
+
+
+   |
+ 9 | {a: 1, **not x, b: 2}
+10 | {**x in y}
+11 | {**x not in y}
+   |      ^^^ Syntax Error: expected Comma, found Not
+12 | {**x < y}
+   |
+
+
+   |
+ 9 | {a: 1, **not x, b: 2}
+10 | {**x in y}
+11 | {**x not in y}
+   |          ^^ Syntax Error: Expected an identifier, but found a keyword 'in' that cannot be used here
+12 | {**x < y}
+   |
+
+
+   |
+ 9 | {a: 1, **not x, b: 2}
+10 | {**x in y}
+11 | {**x not in y}
+   |             ^ Syntax Error: expected Colon, found Name
+12 | {**x < y}
+   |
+
+
+   |
+10 | {**x in y}
+11 | {**x not in y}
+12 | {**x < y}
```

Comment:
This is for the future. We could in theory parse this as a binary expression with a missing left side.

Reply:
The parser will be able to recover well with the star expression update I will be making in a follow-up PR. The grammar here is similar to star expression (`** bitwise_or`) but we'll allow the parser to be lenient and report an error.